api/jobs.go

```diff
@@ -1061,9 +1061,7 @@ func (js *JobSubmission) Canonicalize() {
 	// characters to preserve them. This way, when the job gets stopped and
 	// restarted in the UI, variable values will be parsed correctly.
 	for k, v := range js.VariableFlags {
-		if strings.Contains(v, "\n") {
-			js.VariableFlags[k] = strings.ReplaceAll(v, "\n", "\\n")
-		}
+		js.VariableFlags[k] = url.QueryEscape(v)
```

Comment:
This fixes the Terraform provider and the CLI, because they both use this `api` package, but does the UI already have an equivalent to this fix? (cc @philrenaud )

Should we add a warning to the API documentation for the [`JobSubmission` field](https://developer.hashicorp.com/nomad/api-docs/jobs#submission) that [third-party callers](https://developer.hashicorp.com/nomad/tools#community-tools) of the HTTP API need to perform this operation as well?

Reply:
The UI runs output here through [this jsonToHcl() helper method](https://github.com/hashicorp/nomad/blob/main/ui/app/utils/json-to-hcl.js#L14-L24)

It adds `\n` between multiple VariableFlags / Variables entries, and the [codemirror editor](https://codemirror.net/) we use for rendering generally handles them correctly. It's not perfect, though: Your example in the test, 

```
"test": `"foo": "bar"`
```

renders like this:

![image](https://github.com/user-attachments/assets/390b8061-aa24-4faf-b820-3421867213e7)

which isn't perfect but it's probably enough of an edge case that it's been okay since we've had HCL-in-UI.

============

I'm not sure if this is what url.QueryEscape() is doing, but if we're given output like what's in the test below (%22foo%22%3A+%22bar%22), we will show that verbatim
![image](https://github.com/user-attachments/assets/9e54172b-5b1e-45f3-9bc8-fd88d4aeab97)